src/btree/bt_page.c

```diff
@@ -257,7 +257,9 @@ __wt_page_inmem_prepare(WT_SESSION_IMPL *session, WT_REF *ref)
             }
 
             /* Get the value. */
-            WT_ERR(__wt_page_cell_data_ref(session, page, &unpack, value));
+            WT_ERR(__wt_page_cell_data_ref_kv(session, page, &unpack, value));
+            /* We should never read an overflow removed value for a prepared update. */
+            WT_ASSERT(session, __wt_cell_type_raw(unpack.cell) != WT_CELL_VALUE_OVFL_RM);
```

Comment:
Should this and the subsequent assert be `WT_ASSERT_ALWAYS`? i.e: Should it fail in a release build as well?

Reply:
I can change it to assert always. My main concern is this may impact production performance.

Reply:
Changed to assert always.